app/frontend/translations.ts

```diff
@@ -178,12 +144,70 @@ const translations = {
+  [ExternalErrorReason.PoolRegNotTheOwner]: () =>
+    'This HW device is not an owner of the pool stated in registration certificate.',
+  [ExternalErrorReason.PoolRegInvalidFileFormat]: () =>
+    'Specified file is not a cli-format pool registration certificate transaction.',
+  [ExternalErrorReason.PoolRegWithdrawalDetected]: () =>
+    'The transaction must not include withdrawals.',
+  [ExternalErrorReason.PoolRegInvalidType]: () =>
+    'The certificate in transaction is not a pool registration.',
+  [ExternalErrorReason.PoolRegInvalidNumCerts]: () =>
+    'The transaction must include exactly one certificate, being the pool registration.',
+  [ExternalErrorReason.TrezorRejected]: () =>
+    'TrezorRejected: Operation rejected by the Trezor hardware wallet.',
+  [ExternalErrorReason.DaedalusMnemonic]: () => '',
+
+  [ExternalErrorReason.TransportOpenUserCancelled]: ({message}) => {
+    const errors = {
+      'navigator.usb is undefined':
+        'Your browser does not support WebUSB, use e.g. Google Chrome instead.',
+    }
+
+    return `TransportCanceledByUser: ${message}. ${errors[message] || ''}`
+  },
+  [ExternalErrorReason.TransportError]: ({message}) =>
+    `TransportError: ${message}.If you are using a hardware wallet, please make sure you are using the latest version of the Cardano application.`,
+  [ExternalErrorReason.TransportStatusError]: ({message}) => {
+    const errors = {
+      'Ledger device: Wrong Ledger app':
+        'Please make sure that the Cardano Ledger App is opened before initiating the connection.',
+      'Ledger device: Device is locked': 'Please unlock your device.',
+    }
+    return `TransportStatusError: ${message}. ${errors[message] || ''}`
+  },
+  [ExternalErrorReason.TransportInterfaceNotAvailable]: ({message}) => {
+    const errors = {
+      'Unable to claim interface.':
+        'Please make sure that no other web page/app is interacting with your Ledger device at the same time.',
+    }
+    return `TransportInterfaceNotAvailable: ${message} ${errors[message] || ''}`
+  },
+  [ExternalErrorReason.DisconnectedDeviceDuringOperation]: () =>
+    `DisconnectedDeviceDuringOperation: ${ledgerTroubleshootingSuggestion}`,
+  [ExternalErrorReason.TransportWebUSBGestureRequired]: () =>
+    `TransportWebUSBGestureRequired: ${ledgerTroubleshootingSuggestion}`,
+  [ExternalErrorReason.NotFoundError]: () => `NotFoundError: ${ledgerTroubleshootingSuggestion}`,
+  [ExternalErrorReason.AbortError]: () => `NotFoundError: ${ledgerTroubleshootingSuggestion}`,
+  [ExternalErrorReason.SecurityError]: () => `Access denied: ${ledgerTroubleshootingSuggestion}`,
+  [ExternalErrorReason.RedundantStakePool]: () => 'This stake pool is already chosen.',
+  [ExternalErrorReason.DelegationFeeError]: () => 'Unsuccessful delegation fee calculation.',
+}
+
+function getTranslation(code: InternalErrorReason | ExternalErrorReason, params = {}) {
+  if (InternalErrorReason[code]) {
```

Comment:
I think this maps the errors correctly only thanks to the ExternalErrorReason being declared as a string enum, otherwise, as an integer-based enum it would clash with InternalErrorReason enum

Reply:
If we refactor the InternalError and ExternalError to classes inheriting from a NamedError (arguably something like BaseError would be more suitable of a name then though), we can differentiate here based on "typeof" statement (provided the getTranslation API is refactored to receive the error class itself, which should be straightforward)

Reply:
I think this points to inherent issues with relying on ErrorReason enums - they just won't mix well as enums don't have an explicit type hence are prone to collisions in logic that deals with multiple enums like this one - we discussed this with @PeterBenc and concluded that it's worth wrapping the InternalErrorReason/UnexpectedErrorReason/etc. in proper classes to be able to properly differentiate them in the logic and make ExternalErrorReason a mapping instead of an enum, as it isn't really an exhaustive set of all external errors, only the known ones and should be apprroached as such to avoid confusion/bugs